### DISCUSSION TITLE
Wallet can't connect to Raspiblitz

### DISCUSSION THREAD

**Discussion by Nerfzooka** [raspiblitz/raspiblitz#4448] [Q&A] [answered]
### TLDR:
On a fresh install, wallets can't connect to Raspiblitz. I checked the Electrum connection instructions very closely a number of times, and it worked before the install but the wallets now say "no connection". I can relay some logs, but I can't make sense of them.  Any speculation is welcome.  Thanks.

Wall of text follows:

##### STATUS ELECTRS SERVICE
version='v0.10.0'
configured=1
serviceInstalled=1
serviceRunning=1
localIP='REDACTED'
publicIP='REDACTED'
portTCP='50001'
localTCPPortActive=1
publicTCPPortAnswering=0
portSSL='50002'
localHTTPPortActive=1
publicHTTPPortAnswering=0
TorRunning=1
nginxTest=1
serviceRunning=1
electrumResponding=1
initialSynced=1

Press ENTER to get back to main menu.

***************************************************************
* RASPIBLITZ DEBUG LOGS 
***************************************************************
blitzversion: 1.10.0
chainnetwork: bitcoin / main
 19:07:05 up 2 days,  2:02,  1 user,  load average: 2.66, 2.31, 2.35

*** FAILED SERVICES ***
list any servcies with problems: sudo systemctl list-units --failed
  UNIT LOAD ACTIVE SUB DESCRIPTION
0 loaded units listed.

*** SETUPPHASE / BOOTSTRAP ***
see logs: cat /home/admin/raspiblitz.log
setupPhase--> done
state--> ready

*** BACKGROUNDSERVICE ***
to monitor Background service call: sudo journalctl -f -u background

*** BLOCKCHAIN (MAINNET) SYSTEMD STATUS ***
● bitcoind.service - Bitcoin daemon on mainnet
     Loaded: loaded (/etc/systemd/system/bitcoind.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2024-02-27 17:07:38 GMT; 2 days ago
    Process: 6585 ExecStartPre=/home/admin/config.scripts/bitcoin.check.sh prestart mainnet (code=exited, status=0/SUCCESS)
    Process: 6603 ExecStart=/usr/local/bin/bitcoind -daemonwait -conf=/mnt/hdd/bitcoin/bitcoin.conf -datadir=/mnt/hdd/bitcoin (code=exited, status=0/SUCCESS)
   Main PID: 6647 (bitcoind)
      Tasks: 18 (limit: 4531)
        CPU: 1d 2h 38min 16.337s
     CGroup: /system.slice/bitcoind.service
             └─6647 /usr/local/bin/bitcoind -daemonwait -conf=/mnt/hdd/bitcoin/bitcoin.conf -datadir=/mnt/hdd/bitcoin

Feb 27 17:06:18 raspberrypi systemd[1]: Starting Bitcoin daemon on mainnet...
Feb 27 17:07:38 raspberrypi systemd[1]: Started Bitcoin daemon on mainnet.

*** LAST BLOCKCHAIN (MAINNET) ERROR LOGS ***
sudo journalctl -u bitcoind -b --no-pager -n20
-- Journal begins at Sun 2024-02-25 04:12:28 GMT, ends at Thu 2024-02-29 19:07:05 GMT. --
Feb 27 17:06:18 raspberrypi systemd[1]: Starting Bitcoin daemon on mainnet...
Feb 27 17:07:38 raspberrypi systemd[1]: Started Bitcoin daemon on mainnet.

*** LAST BLOCKCHAIN (MAINNET) INFO LOGS ***
sudo tail -n 50 /mnt/hdd/bitcoin/debug.log
2024-02-29T17:54:57Z New outbound peer connected: version: 70016, blocks=832564, peer=2131 (block-relay-only)
2024-02-29T17:55:59Z New outbound peer connected: version: 70016, blocks=832564, peer=2132 (block-relay-only)
2024-02-29T17:57:09Z Saw new header hash=00000000000000000001b8ddafaf386ce57a229fb95bbc783ce9ddfb9cf427f3 height=832565
2024-02-29T17:57:09Z [net] Saw new cmpctblock header hash=00000000000000000001b8ddafaf386ce57a229fb95bbc783ce9ddfb9cf427f3 peer=1965
2024-02-29T17:57:17Z UpdateTip: new best=00000000000000000001b8ddafaf386ce57a229fb95bbc783ce9ddfb9cf427f3 height=832565 version=0x25fdc000 log2_work=94.763931 tx=971093929 date='2024-02-29T17:56:48Z' progress=1.000000 cache=33.6MiB(87885txo)
2024-02-29T17:59:00Z New outbound peer connected: version: 70016, blocks=832565, peer=2135 (block-relay-only)
2024-02-29T17:59:24Z Saw new header hash=0000000000000000000320ae221fdeb2b8dc44d4b83f8c20d2a62b64ad616764 height=832566
2024-02-29T17:59:24Z [net] Saw new cmpctblock header hash=0000000000000000000320ae221fdeb2b8dc44d4b83f8c20d2a62b64ad616764 peer=1965
2024-02-29T17:59:47Z UpdateTip: new best=0000000000000000000320ae221fdeb2b8dc44d4b83f8c20d2a62b64ad616764 height=832566 version=0x26000000 log2_work=94.763946 tx=971097265 date='2024-02-29T17:58:59Z' progress=1.000000 cache=35.3MiB(101152txo)
2024-02-29T17:59:47Z BlockUntilSyncedToCurrentChain: txindex is catching up on block notifications
2024-02-29T18:04:55Z Socks5() connect to ***.onion:8333 failed: host unreachable
2024-02-29T18:05:53Z Saw new header hash=000000000000000000035b42d1c72ee66f38eee9d870ac4d9953711c78b4be0e height=832567
2024-02-29T18:05:53Z [net] Saw new cmpctblock header hash=000000000000000000035b42d1c72ee66f38eee9d870ac4d9953711c78b4be0e peer=1965
2024-02-29T18:06:06Z UpdateTip: new best=000000000000000000035b42d1c72ee66f38eee9d870ac4d9953711c78b4be0e height=832567 version=0x20800000 log2_work=94.763961 tx=971100880 date='2024-02-29T18:05:26Z' progress=1.000000 cache=37.1MiB(114595txo)
2024-02-29T18:09:00Z New outbound peer connected: version: 70016, blocks=832567, peer=2141 (block-relay-only)
2024-02-29T18:11:35Z Saw new header hash=000000000000000000025a8e4e6067bd4b1e970cb3a815a6514ebc9f4821b8ca height=832568
2024-02-29T18:11:35Z [net] Saw new cmpctblock header hash=000000000000000000025a8e4e6067bd4b1e970cb3a815a6514ebc9f4821b8ca peer=523
2024-02-29T18:12:01Z UpdateTip: new best=000000000000000000025a8e4e6067bd4b1e970cb3a815a6514ebc9f4821b8ca height=832568 version=0x3fffe000 log2_work=94.763976 tx=971104125 date='2024-02-29T18:11:06Z' progress=1.000000 cache=39.2MiB(131370txo)
2024-02-29T18:12:06Z New outbound peer connected: version: 70016, blocks=832568, peer=2144 (block-relay-only)
2024-02-29T18:14:44Z New outbound peer connected: version: 70016, blocks=832568, peer=2149 (block-relay-only)
2024-02-29T18:16:22Z New outbound peer connected: version: 70016, blocks=832568, peer=2153 (block-relay-only)
2024-02-29T18:18:52Z New outbound peer connected: version: 70016, blocks=832568, peer=2156 (block-relay-only)
2024-02-29T18:21:39Z Saw new header hash=00000000000000000000b2d58715cb8df2c4196edf21397fbdd46f9ec16be3ae height=832569
2024-02-29T18:21:39Z [net] Saw new cmpctblock header hash=00000000000000000000b2d58715cb8df2c4196edf21397fbdd46f9ec16be3ae peer=1965
2024-02-29T18:21:45Z UpdateTip: new best=00000000000000000000b2d58715cb8df2c4196edf21397fbdd46f9ec16be3ae height=832569 version=0x34f76000 log2_work=94.763991 tx=971107368 date='2024-02-29T18:21:25Z' progress=1.000000 cache=41.1MiB(146203txo)
2024-02-29T18:21:46Z New outbound peer connected: version: 70016, blocks=832569, peer=2158 (block-relay-only)
2024-02-29T18:25:30Z Saw new header hash=000000000000000000029bf5a15834627339b4e94cef1ca9df2e8caed631fd1c height=832570
2024-02-29T18:25:30Z [net] Saw new cmpctblock header hash=000000000000000000029bf5a15834627339b4e94cef1ca9df2e8caed631fd1c peer=1965
2024-02-29T18:26:04Z UpdateTip: new best=000000000000000000029bf5a15834627339b4e94cef1ca9df2e8caed631fd1c height=832570 version=0x2b264000 log2_work=94.764006 tx=971110580 date='2024-02-29T18:25:10Z' progress=1.000000 cache=42.7MiB(158388txo)
2024-02-29T18:26:05Z Saw new header hash=000000000000000000014750f9e913dfaa23f07737b41542234a4991733c4256 height=832571
2024-02-29T18:26:05Z [net] Saw new cmpctblock header hash=000000000000000000014750f9e913dfaa23f07737b41542234a4991733c4256 peer=1965
2024-02-29T18:26:05Z New outbound peer connected: version: 70016, blocks=832571, peer=2159 (block-relay-only)
2024-02-29T18:26:27Z UpdateTip: new best=000000000000000000014750f9e913dfaa23f07737b41542234a4991733c4256 height=832571 version=0x30aa4000 log2_work=94.764021 tx=971114660 date='2024-02-29T18:25:30Z' progress=1.000000 cache=44.2MiB(169404txo)
2024-02-29T18:26:27Z BlockUntilSyncedToCurrentChain: txindex is catching up on block notifications
2024-02-29T18:26:27Z BlockUntilSyncedToCurrentChain: txindex is catching up on block notifications
2024-02-29T18:26:27Z BlockUntilSyncedToCurrentChain: txindex is catching up on block notifications
2024-02-29T18:28:38Z CreateNewBlock(): block weight: 3992715 txs: 3940 fees: 55300301 sigops 12921
2024-02-29T18:32:30Z New outbound peer connected: version: 70016, blocks=823906, peer=2167 (block-relay-only)
2024-02-29T18:40:00Z New outbound peer connected: version: 70016, blocks=832571, peer=2172 (block-relay-only)
2024-02-29T18:42:10Z Socks5() connect to ***.onion:8333 failed: host unreachable
2024-02-29T18:42:17Z Socks5() connect to ***.onion:8333 failed: host unreachable
2024-02-29T18:44:41Z Saw new header hash=00000000000000000003244593fc89d2dea63c7cc5d13c8fc2f4237c53ad0aba height=832572
_____________________________________________________________________________________
###  **"sudo journalctl -u electrs" output**
-- Journal begins at Sun 2024-02-25 04:12:28 GMT, ends at Thu 2024-02-29 19:18:31 GMT. --
Feb 26 15:35:42 raspberrypi systemd[1]: Started Electrs.
Feb 26 15:35:42 raspberrypi electrs[3829169]: Starting electrs 0.10.0 on aarch64 linux with Config { network: Bitcoin, db_path: "/mnt/hdd/app-storage/electrs/db/bitcoin", daemon_dir: "/home/electrs/.bitcoin", daemon_auth: UserPass("raspibolt", "<sensitive>"), daemon_rpc_addr: 127.0.0.1:8332, daemon_p2p_addr: 127.0.0.1:8333, electrum_rpc_addr: 0.0.0.0:50001, monitoring_addr: 127.0.0.1:4224, wait_duration: 10s, jsonrpc_timeout: 15s, index_batch_size: 10, index_lookup_limit: None, reindex_last_blocks: 0, auto_reindex: true, ignore_mempool: false, sync_once: false, skip_block_download_wait: false, disable_electrum_rpc: false, server_banner: "Welcome to electrs v0.10.0 - the Electrum Rust Server on your RaspiBlitz", signet_magic: f9beb4d9, args: [] }
Feb 26 15:51:14 raspberrypi electrs[3829169]: [2024-02-26T15:51:14.421Z ERROR electrs::server] 9: disconnecting due to failed to send response: "{\"error\":{\"code\":-32603,\"message\":\"unavailable index\"},\"id\":\"electrs-check\",\"jsonrpc\":\"2.0\"}\n"
Feb 26 15:51:14 raspberrypi electrs[3829169]: [2024-02-26T15:51:14.422Z WARN  electrs::server] 9: failed to shutdown TCP connection Transport endpoint is not connected (os error 107)
Feb 26 17:35:22 raspberrypi electrs[3829169]: [2024-02-26T17:35:22.690Z ERROR electrs::server] 493: disconnecting due to failed to send response: "{\"error\":{\"code\":-32603,\"message\":\"unavailable index\"},\"id\":\"electrs-check\",\"jsonrpc\":\"2.0\"}\n"
Feb 26 17:35:22 raspberrypi electrs[3829169]: [2024-02-26T17:35:22.691Z WARN  electrs::server] 493: failed to shutdown TCP connection Transport endpoint is not connected (os error 107)
Feb 27 17:00:56 raspberrypi electrs[3829169]: Error: electrs failed
Feb 27 17:00:56 raspberrypi electrs[3829169]: Caused by:
Feb 27 17:00:56 raspberrypi electrs[3829169]:     0: sync failed
Feb 27 17:00:56 raspberrypi electrs[3829169]:     1: sending on a disconnected channel
Feb 27 17:00:58 raspberrypi systemd[1]: electrs.service: Main process exited, code=exited, status=1/FAILURE
Feb 27 17:00:58 raspberrypi systemd[1]: electrs.service: Failed with result 'exit-code'.
Feb 27 17:00:58 raspberrypi systemd[1]: electrs.service: Consumed 21h 11min 8.247s CPU time.
Feb 27 17:01:58 raspberrypi systemd[1]: electrs.service: Scheduled restart job, restart counter is at 1.
Feb 27 17:01:58 raspberrypi systemd[1]: Stopped Electrs.
Feb 27 17:01:58 raspberrypi systemd[1]: electrs.service: Consumed 21h 11min 8.247s CPU time.
Feb 27 17:02:39 raspberrypi systemd[1]: Started Electrs.
Feb 27 17:02:40 raspberrypi electrs[2594922]: Starting electrs 0.10.0 on aarch64 linux with Config { network: Bitcoin, db_path: "/mnt/hdd/app-storage/electrs/db/bitcoin", daemon_dir: "/home/electrs/.bitcoin", daemon_auth: UserPass("raspibolt", "<sensitive>"), daemon_rpc_addr: 127.0.0.1:8332, daemon_p2p_addr: 127.0.0.1:8333, electrum_rpc_addr: 0.0.0.0:50001, monitoring_addr: 127.0.0.1:4224, wait_duration: 10s, jsonrpc_timeout: 15s, index_batch_size: 10, index_lookup_limit: None, reindex_last_blocks: 0, auto_reindex: true, ignore_mempool: false, sync_once: false, skip_block_download_wait: false, disable_electrum_rpc: false, server_banner: "Welcome to electrs v0.10.0 - the Electrum Rust Server on your RaspiBlitz", signet_magic: f9beb4d9, args: [] }
Feb 27 17:04:26 raspberrypi electrs[2594922]: Error: electrs failed
Feb 27 17:04:26 raspberrypi electrs[2594922]: Caused by:
Feb 27 17:04:26 raspberrypi electrs[2594922]:     0: sync failed
Feb 27 17:04:26 raspberrypi electrs[2594922]:     1: sending on a disconnected channel
Feb 27 17:04:26 raspberrypi systemd[1]: electrs.service: Main process exited, code=exited, status=1/FAILURE
Feb 27 17:04:26 raspberrypi systemd[1]: electrs.service: Failed with result 'exit-code'.
Feb 27 17:04:26 raspberrypi systemd[1]: electrs.service: Consumed 14.589s CPU time.
Feb 27 17:04:32 raspberrypi systemd[1]: Stopped Electrs.
Feb 27 17:04:32 raspberrypi systemd[1]: electrs.service: Consumed 14.589s CPU time.
-- Boot 6afc180879904ba0afcb8730084f50f4 --
Feb 27 17:07:38 raspberrypi systemd[1]: Started Electrs.
Feb 27 17:07:39 raspberrypi electrs[12277]: Starting electrs 0.10.0 on aarch64 linux with Config { network: Bitcoin, db_path: "/mnt/hdd/app-storage/electrs/db/bitcoin", daemon_dir: "/home/electrs/.bitcoin", daemon_auth: UserPass("raspibolt", "<sensitive>"), daemon_rpc_addr: 127.0.0.1:8332, daemon_p2p_addr: 127.0.0.1:8333, electrum_rpc_addr: 0.0.0.0:50001, monitoring_addr: 127.0.0.1:4224, wait_duration: 10s, jsonrpc_timeout: 15s, index_batch_size: 10, index_lookup_limit: None, reindex_last_blocks: 0, auto_reindex: true, ignore_mempool: false, sync_once: false, skip_block_download_wait: false, disable_electrum_rpc: false, server_banner: "Welcome to electrs v0.10.0 - the Electrum Rust Server on your RaspiBlitz", signet_magic: f9beb4d9, args: [] }
Feb 28 00:56:41 raspberrypi electrs[12277]: [2024-02-28T00:56:41.274Z WARN  electrs::mempool] mempool sync failed: failed to get mempool txids
Feb 28 03:15:45 raspberrypi electrs[12277]: [2024-02-28T03:15:45.331Z WARN  electrs::mempool] mempool sync failed: failed to get mempool txids
Feb 28 03:22:39 raspberrypi electrs[12277]: [2024-02-28T03:22:39.286Z WARN  electrs::mempool] mempool sync failed: failed to get mempool txids
Feb 28 08:07:49 raspberrypi electrs[12277]: [2024-02-28T08:07:49.043Z WARN  electrs::mempool] mempool sync failed: failed to get mempool txids
Feb 28 09:54:42 raspberrypi electrs[12277]: [2024-02-28T09:54:42.614Z WARN  electrs::mempool] mempool sync failed: failed to get mempool txids
Feb 28 22:18:26 raspberrypi electrs[12277]: [2024-02-28T22:18:26.995Z WARN  electrs::mempool] mempool sync failed: failed to get mempool txids

**Comment by Nerfzooka** [comment, marked as answer]
Electrum wallet used to connect according to the instructions onscreen in Electrum Rust Server "CONNECT How to Connect" where the port is 50002.  After the fresh install this is not connecting.  Roaming around the [electrs github issues](url) https://github.com/romanz/electrs/issues/961 I saw them using port 50001:t and tried it.  It connects now! 😄 

I wonder why ❓

**Comment by rootzoll** [comment]
Just checked on my dev machine with v1.11.0rc3 .. and its still port 5002 thats working. So not sure what is different with your install.

**Comment by jturning** [comment]
Just had this issue. The Electrum Android wallet won't update the cert, so I wiped the app data and cache and reinstalled per another thread I saw. And then it worked fine with 50002:s. My Pi 5 previously was a manually setup Bitcoin node before I tried Raspiblitz. If you do this, transfer funds out or make sure you have the wallet seed before you wipe it.
